packages/dds/matrix/src/test/memory/sharedMatrix.spec.ts

```diff
@@ -0,0 +1,278 @@
+							if (this.localMatrix === undefined) {
+								throw new Error("localMatrix is not initialized");
+							}
+
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.insertCols(Math.floor(this.localMatrix.colCount / 2), 1);
+							}
+						}
+
+						beforeIteration(): void {
+							this.localMatrix = createLocalMatrix({
+								id: "testLocalMatrix",
+								size: matrixSize,
+								initialValue: matrixValue,
+							});
+						}
+					})(),
+				);
+
+				// Test the memory usage of the SharedMatrix for inserting a row in the middle for a given number of times.
+				benchmarkMemory(
+					new (class implements IMemoryTestObject {
+						readonly title = `Insert a row in the middle ${count} times`;
+						private localMatrix: SharedMatrix | undefined;
+
+						async run(): Promise<void> {
+							if (this.localMatrix === undefined) {
+								throw new Error("localMatrix is not initialized");
+							}
+
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.insertRows(Math.floor(this.localMatrix.rowCount / 2), 1);
+							}
+						}
+
+						beforeIteration(): void {
+							this.localMatrix = createLocalMatrix({
+								id: "testLocalMatrix",
+								size: matrixSize,
+								initialValue: matrixValue,
+							});
+						}
+					})(),
+				);
+
+				// Test the memory usage of the SharedMatrix for inserting a row and a column in the middle for a given number of times.
+				benchmarkMemory(
+					new (class implements IMemoryTestObject {
+						readonly title = `Insert a row and a column ${count} times`;
+						private localMatrix: SharedMatrix | undefined;
+
+						async run(): Promise<void> {
+							if (this.localMatrix === undefined) {
+								throw new Error("localMatrix is not initialized");
+							}
+
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.insertCols(Math.floor(this.localMatrix.colCount / 2), 1);
+								this.localMatrix.insertRows(Math.floor(this.localMatrix.rowCount / 2), 1);
+							}
+						}
+
+						beforeIteration(): void {
+							this.localMatrix = createLocalMatrix({
+								id: "testLocalMatrix",
+								size: matrixSize,
+								initialValue: matrixValue,
+							});
+						}
+					})(),
+				);
+
+				/**
+				 * Test the memory usage of the SharedMatrix for inserting a column and a row
+				 *and then removing them right away to see if the memory is released.
+				 * Memory usage should be very low for these test cases.
+				 */
+				benchmarkMemory(
+					new (class implements IMemoryTestObject {
+						readonly title =
+							`Insert a row and a column and remove them right away ${count} times`;
+						private localMatrix: SharedMatrix | undefined;
+
+						async run(): Promise<void> {
+							if (this.localMatrix === undefined) {
+								throw new Error("localMatrix is not initialized");
+							}
+
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.insertCols(Math.floor(this.localMatrix.colCount / 2), 1);
```

Comment:
Nit: it would probably be worth saving off the "middle" values we complete here, to avoid having to compute them multiple times.

Reply:
Isn't the middle values change dynamically along with insert-remove?

Reply:
Gotcha. Updated it and it looks more readable